syncer/syncer.go

```diff
@@ -73,7 +77,18 @@ func (s *syncer) initializePeerMap() {
 	s.peerMap.PutPeers(peerStatuses)
 
 	for peerStatus := range s.syncPeerClient.GetPeerStatusUpdateCh() {
+		bestPeer := s.peerMap.BestPeer(nil)
+		peerBlockNum := peerStatus.Number
```

Comment:
Please note current peerMap scans all peers when calling BestPeer. I'm not sure whether comparing between current best peer and new status is necessary here

Reply:
Fixed 217daa8125c7873e78c3e48c79105faa0cabd806